### DISCUSSION TITLE
Pivot Fields in BelongsToMany relation is ignored inside fieldsForDetail

### DISCUSSION THREAD

**Discussion by MuathIbnHassan** [laravel/nova-issues#4627] [Q&A] [answered]
- Laravel Version: 9.21.6
- Nova Version: 4.12.12
- PHP Version: 8.1.8
- Database Driver & Version: Postgresql 14.4
- Operating System and Version: macOS 12.4
- Browser type and version: Microsoft Edge Version 104.0.1278.2 (Official build) Dev (x86_64)
- Reproduction Repository: https://github.com/MuathIbnHassan/nova-pivot-inside-fieldsForDetail-issue

### Description:

When using fieldsForDetail, any pivot fields in any BelongsToMany Relation will be ignored.
If the same relation exists in the fields method, it will be used instead of anything in``` fieldsForDetail```

### Detailed steps to reproduce the issue on a fresh Nova installation:

add BelongsToMany relation with ```->fields()``` inside ```public function fieldsForDetail``` without add it to ```public function fields```

**Comment by crynobone** [comment, marked as answer]
You still need to add `BelongsToMany` to `fields()`, or `fieldsForIndex()` method

**Reply by MuathIbnHassan** [reply]
unless I don't want to update or create the field, the problem is that ->fields()'s callback (public $fieldsCallback) registered in fieldsForDetail will never be called on the detail page

**Reply by crynobone** [reply]
The detail page are for the parent resource, when you want to fetch the content of `BelongsToMany` it is an Index request.

**Reply by MuathIbnHassan** [reply]
yes it will be rendered in the index, but the customization for the index is inside the callback


if I want to customize the parent field (BelongsToMany)'s data I cannot, right? 

also if I use two relations to the same model the pivot data will be used in all of them, I want to specify them because I may have multiple pivots tables between two same models, but with this, I will not be able to specify pivot for them all, am I right?

**Reply by MuathIbnHassan** [reply]
I will try to do a proof of concept for the last point in a new repo, but this is happening right now in my project
